adafruit_register/i2c_bcd_alarm.py

```diff
@@ -81,7 +96,11 @@ def __init__(
         self.weekday_shared = weekday_shared
         self.weekday_start = weekday_start
 
-    def __get__(self, obj, objtype=None):
+    def __get__(
+        self,
+        obj: Optional[I2CDeviceDriver],
```

Comment:
This is probably my ignorance, but is there a reason this can be Optional?

(p.s. The rest looked good just unsure on this part)

Reply:
If `__get__()` is being called from an instance, `obj` is that instance; if it's being called from the class it's `None`:

```python
class GetterClass:
    def __get__(self, obj, objtype):
        if obj:
            print("Instance")
        else:
            print("Class")

class WrapperClass:
    attribute = GetterClass()

instance = WrapperClass()
_ = instance.attribute
# prints "Instance"
_ = WrapperClass.attribute
# prints "Class"
```

Reply:
Not sure why `objtype` is optional though - unless it's a quirk of the CircuitPython implementation, it should be the type regardless of whether is called via instance or class.  But since it's default is `None` (and I don't want to accidentally change it and break anything) I just typed it as `Optional` in the event it really can be `None`.